syntax = "proto3";

package api;
option go_package = "model/api";

service Forecast {
  rpc Prophet (ProphetRequest) returns (ProphetReply);
}

message ProphetRequest {
    repeated SamplePair values = 1;
    double duration = 2;
}

message SamplePair {
    int64 timestamp = 1;
    double value = 2;
}

message ProphetReply {
    repeated Forecasted values = 1;
}

message Forecasted {
    double timestamp = 1;
    double yhat = 2;
    double yhatLower = 3;
    double yhatUpper = 4;
}
